core/pkg/store/query.go

```diff
@@ -0,0 +1,122 @@
+package store
+
+import (
+	"maps"
+	"sort"
+	"strings"
+
+	uuid "github.com/google/uuid"
+	"github.com/open-feature/flagd/core/pkg/model"
+)
+
+// flags table and index constants
+const flagsTable = "flags"
+
+const idIndex = "id"
+const keyIndex = "key"
+const sourceIndex = "source"
+const priorityIndex = "priority"
+const flagSetIdIndex = "flagSetId"
+
+// compound indices; maintain sub-indexes alphabetically; order matters; these must match what's generated in
+const flagSetIdSourceCompoundIndex = flagSetIdIndex + "+" + sourceIndex
+const keySourceCompoundIndex = keyIndex + "+" + sourceIndex
+const flagSetIdKeySourceCompoundIndex = flagSetIdIndex + "+" + keyIndex + "+" + sourceIndex
+
+// flagSetId defaults to a UUID generated at startup to make our queries consistent
+// any flag without a "flagSetId" is assigned this one; it's never exposed externally
+var nilFlagSetId = uuid.New().String()
+
+// A selector represents a set of constraints used to query the store.
+type Selector struct {
+	indexMap map[string]string
+}
+
+// NewSelector creates a new Selector from a selector expression string.
+// For example, to select flags from source "./mySource" and flagSetId "1234", use the expression:
+// "source=./mySource,flagSetId=1234"
+func NewSelector(selectorExpression string) Selector {
+	return Selector{
+		indexMap: expressionToMap(selectorExpression),
+	}
+}
+
+func expressionToMap(selector string) map[string]string {
+	selectorMap := make(map[string]string)
+	if selector == "" {
+		return selectorMap
+	}
+
+	// Split the selector by commas
+	pairs := strings.Split(selector, ",")
```

Comment:
Both `,` and `=` are allowed in the flag keys, so the parsing here will not work as intended for corner cases.

Reply:
I considered this, but wasn't sure how to proceed! Thanks for the eye for detail. ..

Do you have a suggestion? In general there's not much validation here. I can't decide on the best course. I'm not sure I want to implement escape characters or something.

Reply:
@thisthat 

Any thoughts on this?

Reply:
Since we're getting this data as an HTTP Header, I would vote to follow the format specified in [RFC9651](https://www.rfc-editor.org/rfc/rfc9651) or a subset of it since I don't think we will support list 🤔
Hence, we should replace `,` with `;` and disallow `=` in the schema definition of `metadata.flagSetId`.

Reply:
I'm ok with that. And we can enforce this in the json schema so that violation in flagsetid will cause json validation failures. 

@beeme1mr @tangenti WDYT?

Reply:
If I have to choose, I'd go with `;` since it splits parameters. However, I am happy with both options as long as it is removed from being a valid char in the schema. We can discuss this in a follow-up since this PR is already quite big :)

Reply:
In general I think it's not a good practice to mark some chars as special. 

I've considered ANTLR but it seems a bit overkill at the moment, but if we don't have a better option, I'd suggesting using ANTLR